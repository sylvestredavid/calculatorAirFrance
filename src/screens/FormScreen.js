import React from 'react';
import {View, TouchableOpacity, StyleSheet, Text, Image} from 'react-native';
import {TextInput} from 'react-native-paper';
import Message from '../components/Message';
import {formatText, calculPoidTotal, calculKerozeneUtilise} from '../utils';
import {caracteristiquesAvion} from '../utils/avion';

export default class FormScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      nbPassagers: '',
      nbLitreKerozene: '',
      nbKilosBagagesSoute: '',
      nbKilosBagagesCabine: '',
      nbHeuresVol: '',
      message: undefined,
    };
  }

  changeNbPassager = (e) => {
    const val = formatText(e.nativeEvent.text);
    if (+val > caracteristiquesAvion.nbPassagersMax) {
      this.setState({
        nbPassagers: caracteristiquesAvion.nbPassagersMax,
      });
    } else {
      this.setState({
        nbPassagers: val,
      });
    }
  };

  changeNbLitreKerozene = (e) => {
    const val = formatText(e.nativeEvent.text);
    this.setState({
      nbLitreKerozene: val,
    });
  };

  changeNbKilosBagagesCabine = (e) => {
    const val = formatText(e.nativeEvent.text);
    this.setState({
      nbKilosBagagesCabine: val,
    });
  };

  changeNbKilosBagagesSoute = (e) => {
    const val = formatText(e.nativeEvent.text);
    this.setState({
      nbKilosBagagesSoute: val,
    });
  };

  changeNbHeuresVol = (e) => {
    const val = formatText(e.nativeEvent.text);
    this.setState({
      nbHeuresVol: val,
    });
  };

  onSubmit = () => {
    if (
      this.state.nbPassagers === '' ||
      this.state.nbLitreKerozene === '' ||
      this.state.nbKilosBagagesCabine === '' ||
      this.state.nbKilosBagagesSoute === '' ||
      this.state.nbHeuresVol === ''
    ) {
      this.setState({
        message: {
          style: 'danger',
          text: 'Merci de remplir tout les champs',
        },
      });
    } else {
      const poidTotal = calculPoidTotal(this.state);
      const kerozeneUtilise = calculKerozeneUtilise(this.state.nbHeuresVol);
      if (
        poidTotal <= caracteristiquesAvion.poidMaxKG &&
        kerozeneUtilise <= this.state.nbLitreKerozene
      ) {
        this.setState({
          message: {
            style: 'success',
            text: 'vous pouvez voler! Bon vol!',
          },
        });
      } else {
        if (poidTotal > caracteristiquesAvion.poidMaxKG) {
          this.setState({
            message: {
              style: 'danger',
              text: "l'avion ne peut pas décoller car trop lourd",
            },
          });
        } else if (kerozeneUtilise > this.state.nbLitreKerozene) {
          this.setState({
            message: {
              style: 'danger',
              text:
                "Le temps de vol est supérieur à l'autonomie de l’avion en vue du kérosène embarqué",
            },
          });
        }
      }
    }
  };

  closeMessage = () => {
    this.setState({
      message: undefined,
    });
  };

  render() {
    return (
      <View style={styles.container}>
        <Image
          style={styles.image}
          source={require('../assets/logoAirFrance.png')}
        />
        <View style={styles.form}>
          <TextInput
            mode={'outlined'}
            keyboardType={'number-pad'}
            style={{marginVertical: 5}}
            label={'Passagers'}
            value={this.state.nbPassagers}
            onChange={this.changeNbPassager}
          />
          <TextInput
            mode={'outlined'}
            keyboardType={'number-pad'}
            style={{marginVertical: 5}}
            label={'Kérozene (L)'}
            value={this.state.nbLitreKerozene}
            onChange={this.changeNbLitreKerozene}
          />
          <TextInput
            mode={'outlined'}
            keyboardType={'number-pad'}
            style={{marginVertical: 5}}
            label={'Bagages cabine (Kg)'}
            value={this.state.nbKilosBagagesCabine}
            onChange={this.changeNbKilosBagagesCabine}
          />
          <TextInput
            mode={'outlined'}
            keyboardType={'number-pad'}
            style={{marginVertical: 5}}
            label={'Bagages soute (Kg)'}
            value={this.state.nbKilosBagagesSoute}
            onChange={this.changeNbKilosBagagesSoute}
          />
          <TextInput
            mode={'outlined'}
            keyboardType={'number-pad'}
            style={{marginVertical: 5}}
            label={'Heures de vol'}
            value={this.state.nbHeuresVol}
            onChange={this.changeNbHeuresVol}
          />
        </View>
        {this.state.message && (
          <Message
            style={this.state.message.style}
            text={this.state.message.text}
            close={this.closeMessage}
          />
        )}
        <TouchableOpacity style={styles.button} onPress={this.onSubmit}>
          <Text style={{color: 'white', fontWeight: 'bold', fontSize: 18}}>
            Calculer
          </Text>
        </TouchableOpacity>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  form: {
    width: '70%',
  },
  container: {
    width: '90%',
    padding: 10,
    marginHorizontal: '5%',
    marginVertical: 20,
    borderWidth: 1,
    borderStyle: 'solid',
    borderColor: 'black',
    position: 'relative',
  },
  button: {
    position: 'absolute',
    bottom: -20,
    right: 10,
    borderWidth: 1,
    borderStyle: 'solid',
    borderColor: 'black',
    backgroundColor: 'red',
    padding: 5,
    borderRadius: 5,
  },
  image: {
    width: '40%',
    height: 20,
    position: 'absolute',
    top: -10,
    right: 10,
  },
});
