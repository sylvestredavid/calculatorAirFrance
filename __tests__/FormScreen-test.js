/**
 * @format
 */

import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import FormScreen from '../src/screens/FormScreen';

test('renders correctly', () => {
  const tree = renderer.create(<FormScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
