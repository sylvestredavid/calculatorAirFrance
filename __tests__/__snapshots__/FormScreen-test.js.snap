// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "black",
      "borderStyle": "solid",
      "borderWidth": 1,
      "marginHorizontal": "5%",
      "marginTop": 20,
      "padding": 10,
      "position": "relative",
      "width": "90%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/logoAirFrance.png",
      }
    }
    style={
      Object {
        "height": 20,
        "position": "absolute",
        "right": 10,
        "top": -10,
        "width": "40%",
      }
    }
  />
  <View
    style={
      Object {
        "width": "70%",
      }
    }
  >
    <View
      style={
        Object {
          "marginVertical": 5,
        }
      }
    >
      <View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 8,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "bottom": 0,
                  "height": 2,
                  "left": 10,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 8,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "opacity": 0,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 37,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Passagers
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Passagers
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 1,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Passagers
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            keyboardType="number-pad"
            multiline={false}
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "height": 56,
                },
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 5,
        }
      }
    >
      <View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 8,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "bottom": 0,
                  "height": 2,
                  "left": 10,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 8,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "opacity": 0,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 37,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              KÃ©rozene (L)
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              KÃ©rozene (L)
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 1,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              KÃ©rozene (L)
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            keyboardType="number-pad"
            multiline={false}
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "height": 56,
                },
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 5,
        }
      }
    >
      <View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 8,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "bottom": 0,
                  "height": 2,
                  "left": 10,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 8,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "opacity": 0,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 37,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Bagages cabine (Kg)
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Bagages cabine (Kg)
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 1,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Bagages cabine (Kg)
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            keyboardType="number-pad"
            multiline={false}
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "height": 56,
                },
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 5,
        }
      }
    >
      <View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 8,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "bottom": 0,
                  "height": 2,
                  "left": 10,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 8,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "opacity": 0,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 37,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Bagages soute (Kg)
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Bagages soute (Kg)
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 1,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Bagages soute (Kg)
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            keyboardType="number-pad"
            multiline={false}
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "height": 56,
                },
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 5,
        }
      }
    >
      <View>
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              Object {
                "backgroundColor": "#f6f6f6",
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderRadius": 4,
                "borderWidth": 1,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 8,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "bottom": 0,
                  "height": 2,
                  "left": 10,
                  "opacity": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": 8,
                }
              }
            />
            <Text
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#f6f6f6",
                  "color": "transparent",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 18,
                  "opacity": 0,
                  "paddingHorizontal": 0,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 37,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "scaleY": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Heures de vol
            </Text>
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "#6200ee",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 0,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Heures de vol
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "left": 0,
                  "opacity": 1,
                  "paddingHorizontal": 14,
                  "position": "absolute",
                  "textAlign": "left",
                  "top": 36,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                    Object {
                      "scale": 1,
                    },
                  ],
                  "writingDirection": "ltr",
                }
              }
            >
              Heures de vol
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            editable={true}
            keyboardType="number-pad"
            multiline={false}
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder=""
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "paddingHorizontal": 14,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "height": 56,
                },
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "borderColor": "black",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 1,
        "bottom": -20,
        "opacity": 1,
        "padding": 5,
        "position": "absolute",
        "right": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Calculer
    </Text>
  </View>
</View>
`;
